enumerable.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(i, self[i])
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    my_each do |x|
+      new_array << x if yield(x)
+    end
+    new_array
+  end
+
+  def my_all?
+    answer = true
+    my_each do |x|
+      answer = false unless yield(x)
+    end
+    answer
+  end
+
+  def my_any?
+    answer = false
+    my_each do |x|
+      answer = true if yield(x)
+    end
+    answer
+  end
+
+  def my_none?
+    answer = true
+    my_each do |x|
+      answer = false if yield(x)
+    end
+    answer
+  end
+
+  def my_count
+    counter = 0
+    my_each do |x|
+      counter += 1 if yield(x)
+    end
+    counter
+  end
+
+  def my_map(param = nil)
+    new_array = []
+    if block_given?
+      my_each do |x|
+        new_array << yield(x)
+      end
+    else
+      my_each do |x|
+        new_array << param.call(x)
+      end
+    end
+    new_array
+  end
+
+  def my_inject
+    answer = yield(0, 1).zero? ? 1 : 0
+    my_each do |x|
+      answer = yield(answer, x)
+    end
+    answer
+  end
+
+  def multiply_els
+    my_inject { |x, y| x * y }
+  end
+end
```

Comment:
* [ ] the method should be able to take literals, to_proc symbols(e.g :+, :-, etc... ) as arguments, yours raise an error when doing so red_circle

```ruby
range = Range.new(5,50)
range.my_inject(4) { |prod, n| prod * n }
      # return this error
      # ArgumentError:
      #  wrong number of arguments (given 1, expected 0)
      # ./enumerable.rb:74:in `my_inject'
```